scripts/spm-integration-test.sh

```diff
@@ -61,13 +61,18 @@ validate_service_metrics() {
     # Store the values in an array
     mapfile -t metric_points < <(echo "$response" | jq -r '.metrics[0].metricPoints[].gaugeValue.doubleValue')
     echo "Metric datapoints found for service '$service': " "${metric_points[@]}"
-    # Check that all values are non-zero
+    # Check that atleast some values are non-zero after the threshold
     local non_zero_count=0
+    local threshold=3
```

Comment:
```suggestion
    local non_zero_count=0
    local expected_non_zero_count=3
    local zero_count=0
    local expected_max_zero_count=3
```

Reply:
but tbh, I would consider deleting zero_count logic altogether. If zero values can be produced due to race conditions in the start-up, maybe we should only be looking for 3+ non-zero values and not worry about zeros.

If we still want to check for errors with a threshold, then it would make sense to only allow zeros at the beginning on the time series, once we start getting data I don't see a strong reason why zeros should show up again

Reply:
Yea makes sense, So I will check until I get a non-zero value or it reaches threshold. Does this work ?

Reply:
Also maybe wait after you get the first non-zero value ?

Reply:
we already have the timeout logic handled. This function just checks if "in the current state" the data satisfies the requirements. And that requirement could be `0{0-3} >0{3+}`, i.e. no more than 3 leading zeroes followed by at least 3 non-zeros.

Reply:
Understood. I will make the changes.

Reply:
@FlamingSaint I merged this for now to prevent spurious failures affecting other PRs, but let's make the change we discussed as a next improvement.